src/google/protobuf/compiler/command_line_interface.cc

```diff
@@ -188,8 +188,25 @@ bool TryCreateParentDirectory(const string& prefix, const string& filename) {
 // Get the absolute path of this protoc binary.
 bool GetProtocAbsolutePath(string* path) {
 #ifdef _WIN32
-  char buffer[MAX_PATH];
-  int len = GetModuleFileName(NULL, buffer, MAX_PATH);
+  wchar_t wide_buffer[MAX_PATH];
+  // Note: the returned length does not include terminating null character.
+  int len = GetModuleFileNameW(NULL, wide_buffer, MAX_PATH);
+  if (len == GOOGLE_ARRAYSIZE(wide_buffer) &&
+      GetLastError() == ERROR_INSUFFICIENT_BUFFER) {
+    // The path is too long.
+    return false;
+  }
+  char buffer[MAX_PATH + 1];
+  for (size_t i = 0; i < len; ++i) {
+    wchar_t wide_char = wide_buffer[i];
+    if (wide_char > 0x7F) {
+      // non-ASCII character is not supported yet.
+      // TODO(yukawa): Support paths that contain non-ASCII characters.
```

Comment:
If there is an encoding with which "/includle" is not a valid byte sequence, protocol compiler will definitely not work on such platforms no matter what we do here (e.g., we replace ".proto" with ".pb.h" when generating header files. If ".pb.h" is invalid, ...)

I don't see any problem by letting the binary representation depend on whatever encoding the system is using. Actually I think it's the correct way here. Protocol compiler uses lower-level file APIs to read and write files (e.g., open or fopen). I believe these APIs does not handle character encodings and you have to use char sequences in the system locale to successfully open files with non-ASCII names.